### PR TITLE
[BUG] new.sh fix generator type , issue #9320

### PR DESCRIPTION
**Pull request by AndersSpringborg**
This is a small fix, for generating a new template. It helps getting started.
I found the issue today, as I was making a new generator.

I've made a Issue, describing the problem #9320 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: new.sh [hunk 1]
```diff
@@ -205,7 +205,7 @@ cat > "${root}/bin/configs/${gen_name_camel}-${gen_type}-petstore-new.yaml"<<EOF
 generatorName: ${gen_name_camel}
 outputDir: samples/${gen_type}/petstore/${gen_name_camel_path}
 inputSpec: modules/openapi-generator/src/test/resources/3_0/petstore.yaml
-templateDir: modules/openapi-generator/src/main/resources/${gen_name_camel}
+templateDir: modules/openapi-generator/src/main/resources/${gen_name_camel}-${gen_type}
```
Comment (wing328) [inline new.sh:208]: @AndersSpringborg thanks for the PR but I think we usually use the generator name directly as the template folder name (e.g. lua, perl) without adding the generator type (e.g. client, server)

Comment (AndersSpringborg) [reply new.sh:208]: Oh yeah, i can see what you mean now!

If you follow the instruction on how to make a new generator, it gives an error, that I cannot find that template. I'll look into what courses the problem tomorrow 😊

Comment (AndersSpringborg) [reply new.sh:208]: In the the script, the template folder is created with `${gen_name_camel}-${gen_type}`
https://github.com/OpenAPITools/openapi-generator/blob/0f5e7d1e3cb9642bd46ec89ffe0bf66a922029f2/new.sh#L193-L200

So the template dir in the config, should have the same path
https://github.com/OpenAPITools/openapi-generator/blob/0f5e7d1e3cb9642bd46ec89ffe0bf66a922029f2/new.sh#L208

Comment (wing328) [reply new.sh:208]: Good point. What about updating these lines instead following the convention that I mentioned?
```
# Step 3: Create resource files 
 mkdir -p "${root}/modules/openapi-generator/src/main/resources/${gen_name_camel}-${gen_type}" 
 echo "Creating modules/openapi-generator/src/main/resources/${gen_name_camel}-${gen_type}/README.mustache" && \ 
     touch "${root}/modules/openapi-generator/src/main/resources/${gen_name_camel}-${gen_type}/README.mustache" 
 echo "Creating modules/openapi-generator/src/main/resources/${gen_name_camel}-${gen_type}/model.mustache" && \ 
     touch "${root}/modules/openapi-generator/src/main/resources/${gen_name_camel}-${gen_type}/model.mustache" 
 echo "Creating modules/openapi-generator/src/main/resources/${gen_name_camel}-${gen_type}/api.mustache" && \ 
     touch "${root}/modules/openapi-generator/src/main/resources/${gen_name_camel}-${gen_type}/api.mustache" 
```

Comment (AndersSpringborg) [reply new.sh:208]: Let me know if that's how you want the script to work! I'll go through it, and make sure it works with this naming convention

Comment (wing328) [reply new.sh:208]: Please update based on https://github.com/OpenAPITools/openapi-generator/pull/9321#discussion_r619915054. Thank you.

**Comment by wing328** [comment]
Tested and looks good
```
./new.sh -n julia -c
Creating modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JuliaClientCodegen.java
Creating modules/openapi-generator/src/main/resources/julia/README.mustache
Creating modules/openapi-generator/src/main/resources/julia/model.mustache
Creating modules/openapi-generator/src/main/resources/julia/api.mustache
Creating bin/configs/julia-petstore-new.yaml
Finished.
```

**Pull request merged**